lib/hyrax/specs/shared_specs/indexers.rb

```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 RSpec.shared_examples 'a Hyrax::Resource indexer' do
-  subject(:indexer) { indexer_class.new(resource: resource) }
-  let(:ids)         { ['id1', 'id2'] }
-  let(:resource)    { Hyrax::Resource.new(alternate_ids: ids) }
+  subject(:indexer)  { indexer_class.new(resource: the_resource) }
+  let(:ids)          { ['id1', 'id2'] }
+  let(:the_resource) { defined?(resource) ? resource : Hyrax::Resource.new }
```

Comment:
This is providing backward compatibility.  If the caller does not create a resource, then one is created in the shared spec.  If backward compatibility is not required, this setup would become...

```
   subject(:indexer)  { indexer_class.new(resource: resource) }
   let(:ids)          { ['id1', 'id2'] }
```

NOTE: All references to `the_resource` are providing backward compatibility for each shared spec.  Similar changes could occur if `resource` was required to be defined by the caller.

Reply:
Based on conversations, the reason for these conditionals is for backwards compatibility.

Reply:
As this was not released in a Hyrax version, we can remove the conditionals and tighten up the logic. Anyone that's using this as written will have a broken spec.